### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by wenwenchenbosch**
add coverage report in jenkins and some cleanup of code coverage script.

gcovr_4.2-1 is used for coverage, necessary to clear? But it is BSD licence: https://gcovr.com/en/stable/license.html

### CODE REVIEW

**Comment by SebastianSchildt** [comment]
It seems it needs additonal Jenkins Plugins?

```
java.lang.NoSuchMethodError: No such DSL method 'cobertura' found among steps [acceptGitLabMR, 
```

can you list which dependencies are needed? I think this needs another ticket/PR at Eclipse CI infrastructure and we should get it right first time...

**Comment by wenwenchenbosch** [comment]
> It seems it needs additonal Jenkins Plugins?
> 
> ```
> java.lang.NoSuchMethodError: No such DSL method 'cobertura' found among steps [acceptGitLabMR, 
> ```
> 
> can you list which dependencies are needed? I think this needs another ticket/PR at Eclipse CI infrastructure and we should get it right first time...

This plugin coberatura https://plugins.jenkins.io/cobertura/ is necessary.

File: Jenkinsfile [hunk 1]
```diff
@@ -12,8 +12,10 @@ node('docker') {
     checkout scm
     def versiontag="unknown"
     stage('Prepare') {
+        sh 'ls -al artifacts'
```
Comment (SebastianSchildt) [inline Jenkinsfile:15]: Why?

File: Jenkinsfile [hunk 2]
```diff
@@ -42,14 +44,15 @@ node('docker') {
...
         step([$class: 'XUnitPublisher',
                 thresholds: [ skipped(failureThreshold: '0'), failed(failureThreshold: '0') ],
-                tools: [ BoostTest(pattern: 'artifacts/*.xml') ]]
+                tools: [ BoostTest(pattern: 'artifacts/results.xml') ]]
         )
+        cobertura autoUpdateHealth: false, autoUpdateStability: false, coberturaReportFile: 'artifacts/coverage.xml', conditionalCoverageTargets: '70, 0, 0', enableNewApi: true, failUnhealthy: false, failUnstable: false, lineCoverageTargets: '80, 0, 0', maxNumberOfBuilds: 0, methodCoverageTargets: '80, 0, 0', onlyStable: false, sourceEncoding: 'ASCII', zoomCoverageChart: false
     }
     stage('Compress') {
-        sh 'ls artifacts'
+        sh 'ls -al artifacts'
```
Comment (SebastianSchildt) [inline Jenkinsfile:55]: Why? If useful for debugging, we can keep it, if just forgot to delete it, maybe remove...

File: docker/Dockerfile.dev [hunk 3]
```diff
@@ -28,11 +28,14 @@ RUN DEBIAN_FRONTEND="noninteractive" apt-get -y install libssl-dev  \
                         libboost-thread1.67-dev\
                         libboost-test1.67-dev\
                         libmosquitto-dev
+RUN DEBIAN_FRONTEND="noninteractive" apt-get -y install  \
+                        # workaround, because a lot of packages above does not exists any more
```
Comment (SebastianSchildt) [inline docker/Dockerfile.dev:32]: why is this a workaround? And why not just adding gcovr to the previous call?

**Pull request merged**